### ISSUE TITLE
[FEATURE] Add Tesla Plugin

### ISSUE THREAD

**Issue by yordis** [akoutmos/prom_ex#132]
**Describe the solution you would like to see**

As a user, I want PromEx to include Tesla Plugin.

**How would you expect this feature to work**

Tesla HTTP Middleware client (fair to say popular) has built-in Telemetry middleware that it would be nice to collect with low effort.

**Comment by akoutmos** [comment]
Hello Yordis and thanks for reaching out!

Support for the Tesla telemetry middleware is indeed on the roadmap. Unfortunately, I won't be able to give you a solid ETA as I am currently working on PromEx features requested by some of my consulting clients. If I were to ballpark estimate...I would say I can get to this in the next 3 months or so.

**Comment by akoutmos** [comment]
Good call. I haven't gotten around to making a `CONTRIBUTING.md` file yet, but I'll add that to the TODO list. 

For new PromEx plugins, you will need to create a plugin module that will attach to all of the relevant Telemetry events that the Tesla middleware emits, and you will need to create a Grafana dashboard template to render all those datapoints. A good place to start would be to take a look at some of the existing plugins and dashboards, and follow a consistent style in terms of documentation styling and how the plugin is laid out. Looking at the [Tesla docs](https://hexdocs.pm/tesla/Tesla.Middleware.Telemetry.html), if looks like you'll want to attach to the following Telemetry events:

- `[:tesla, :request, :stop]`
- `[:tesla, :request, :exception]`

You will probably need to create Prometheus distributions for both of those events to capture request time. It will also probably be nice to have the host that the request is directed towards, but that will probably need to be some sort of a config so that all the unique hosts don't create a cardinality problem for Prometheus. It would also be useful to capture the HTTP status code from the request. Lastly, I would make another distribution to capture the response body size (similarly to how it is captured in the Phoenix plugin). Let me know if you have any additional questions or is you require some more clarification. Thanks!

**Comment by brandonchoe** [comment]
Hi, I'd like to grab this issue. I'll have a PR up this weekend.
